detour/mesh.go

```diff
@@ -157,21 +162,41 @@ func (m *NavMesh) SaveToFile(fn string) error {
 		var tileHeader navMeshTileHeader
 		tileHeader.TileRef = m.TileRef(tile)
 		tileHeader.DataSize = tile.DataSize
-		if _, err = tileHeader.WriteTo(f); err != nil {
+		if _, err := tileHeader.WriteTo(w); err != nil {
 			return err
 		}
 		var data []byte = make([]byte, tile.DataSize)
 		// first Serialize the tile header
 		tile.Header.serialize(data)
 		// then the tile itself
 		tile.serialize(data[tile.Header.size():])
-		if _, err = f.Write(data); err != nil {
+		if _, err := w.Write(data); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// Returns binary navigation mesh file size.
+func (m *NavMesh) Size() int32 {
+	var header navMeshSetHeader
+	header.Params = m.Params
+	hSize := header.Size()
+
+	var tilesSize int32 = 0
+	var tileHeader navMeshTileHeader
+	for i := int32(0); i < m.MaxTiles; i++ {
+		tile := &m.Tiles[i]
+		if tile.DataSize == 0 {
+			continue
+		}
+
+		tilesSize += int32(tileHeader.Size()) + tile.DataSize
+	}
+
+	return int32(hSize) + tilesSize
+}
+
```

Comment:
What's the use case for that method?
By using the `WriteTo` method (with the new signature), one also obtains the final file size.
If you're not interested by the data, but just by its size, you could something like

```go

var navMesh *NavMesh
n, err := navMesh.WriteTo(io.discard)
if err {...}
fmt.Println("nav mesh file size:", n)
```
Let me know if that works for you.